### ISSUE TITLE
Introspection crash, `libs\datamodel\connectors\dml\src\model.rs:494:29` (missing PK?)

### ISSUE THREAD

**Issue by Historaj** [prisma/prisma#14403]
Hi Prisma Team! Prisma Migrate just crashed. 

## Command

`db pull`

## Versions
      
| Name        | Version            |
|-------------|--------------------|
| Platform    | windows            | 
| Node        | v16.14.0           | 
| Prisma CLI  | 4.1.0              | 
| Engine      | 8d8414deb360336e4698a65aa45a1fbaf1ce13d8| 

## Error
```
Error: [libs\datamodel\connectors\dml\src\model.rs:494:29] Hi there! We've been seeing this error in our error reporting backend, but cannot reproduce it in our own tests. The problem is that we have a primary key in the model `T_Clients` that uses the column `ConsigneeConfirmationDate` which we for some reason don't have in our internal representation. If you see this, could you please file an issue to https://github.com/prisma/prisma so we can discuss about fixing this. -- Your friendly prisma developers. 
```

**Comment by janpio** [comment]
Can you please share the SQL of your `T_Clients` table @Historaj? Thanks.

**Comment by Historaj** [comment]
CREATE TABLE [dbo].[T_Clients](
	[Address] [varchar](50) NULL,
	[AgreementID] [varchar](50) NULL,
	[BillingType] [varchar](50) NULL,
	[CCEmail] [varchar](100) NULL,
	[City] [varchar](50) NULL,
	[ClientCode] [int] IDENTITY(1,1) NOT NULL,
	[ClientDisplayName] [varchar](50) NULL,
	[ClientName] [varchar](50) NULL,
	[Extension] [varchar](50) NULL,
	[Fax] [varchar](50) NULL,
	[FromEMail] [varchar](50) NULL,
	[HomePhone] [varchar](50) NULL,
	[Others_cell] [varchar](50) NULL,
	[Pager] [varchar](50) NULL,
	[ReportMode] [varchar](50) NULL,
	[State] [varchar](50) NULL,
	[Status] [varchar](50) NULL,
	[ToEmail] [varchar](50) NULL,
	[Type] [varchar](50) NULL,
	[WorkPhone] [varchar](50) NULL,
	[Zip] [varchar](50) NULL,
 CONSTRAINT [PK_T_Clients] PRIMARY KEY CLUSTERED 
(
	[ClientCode] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
) ON [PRIMARY]
GO

**Comment by tomhoule** [comment]
This should be fixed in Prisma 4.2, to be released tomorrow (https://github.com/prisma/prisma-engines/pull/3108)
